db/view/regular_column_transformation.hh

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "column_computation.hh"
+#include "mutation/atomic_cell.hh"
+#include "timestamp.hh"
+#include <type_traits>
+
+// In a basic column_computation defined in column_computation.hh, the
+// compute_value() method is only based on the partition key, and it must
+// return a value. That API has very limited applications - basically the
+// only thing we can implement with it is token_column_computation which
+// we used to create the token column in secondary indexes.
+// The regular_column_transformation base class here is more powerful, but
+// still is not a completely general computation: Its compute_value() virtual
+// method can transform the value read from a single cell of regular column
+// in a row, into a new cell (i.e., a value or deletion, timestamp, and TTL),
+// stored in the structure regular_column_transformation::result.
+//
+// In more details, the assumptions of regular_column_transformation is:
+// 1. compute_value() computes the value based on a *single* column in a
+//    row passed to compute_value().
+//    This assumption means that the value or deletion of the value always
+//    has a single known timestamp (and the value can't be half-missing)
+//    and TTL information. That would not have been possible if we allowed
+//    the computation to depend on multiple columns.
+// 2. The source is a *regular* column in the base table. This means that an
+//    update can modify it (unlike a base-table key column that can't change
+//    in an update) so we may need to read the value before and after the
+//    update, and delete and create view rows.
+// 3. compute_value() can return 1. a value, or 2. "missing" if the column
+//    to be transformed is missing from the given row, or 3. "deleted"
+//    if the row contains a deletion (a tombstone) for the column.
+
+class regular_column_transformation : public column_computation {
+public:
+    struct result {
+        std::optional<bytes> _value;
+        bool _deleted = false; // if _deleted, "value" is unset
+        api::timestamp_type _ts = api::missing_timestamp; // defined if has_value() or is_deleted() but not if is_missing()
+        gc_clock::duration _ttl; // defined if has_value()
+        gc_clock::time_point _expiry; // defined if has_value()
```

Comment:
> This sounds like _value should be of std::optional<std::optional<bytes>> type

Or better, why not turn these variables into functions that'd honor the invariants mentioned in the comments.